### PR TITLE
feat(cli): create CLI package

### PR DESCRIPTION
**Pull request by gund**
- [x] Implement create app command
- [x] Implement `create-oryx` package for `npm init` support
- [x] Add unit tests

### CODE REVIEW

File: libs/platform/core/src/services/feature-options/default-feature-options.service.ts [hunk 1]
```diff
@@ -5,14 +5,16 @@ import {
...
 
   constructor(protected optionsArr = inject(FeatureOptions, null)) {
     this.options =
       optionsArr?.reduce((acc, options) => ({ ...acc, ...options }), {}) ?? {};
   }
 
-  getFeatureOptions(name: string): FeatureOptions {
+  getFeatureOptions<N extends keyof FeatureOptions>(name: N): FeatureOptions[N];
+  getFeatureOptions(name: string): object;
+  getFeatureOptions(name: string): object {
```
Comment (supproduction) [inline libs/platform/core/src/services/feature-options/default-feature-options.service.ts:17]:
```suggestion
  getFeatureOptions(name: string): object {
```

File: libs/template/cli/bin/cli.spec.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { vi } from 'vitest';
```
Comment (supproduction) [inline libs/template/cli/bin/cli.spec.ts:1]: We don't need this import

File: libs/template/cli/package.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  "dependencies": {
+    "@clack/prompts": "^0.6.3",
+    "picocolors": "^1.0.0",
+    "request": "^2.88.2",
+    "rxjs": "^7.5.0",
+    "unzipper": "^0.10.11"
+  },
+  "peerDependencies": {
+    "@spryker-oryx/core": "^0.4.0",
+    "@spryker-oryx/di": "^0.4.0"
```
Comment (supproduction) [inline libs/template/cli/package.json]: Is it ok that in other packages we use `latest` ?

File: libs/template/cli/src/commands/create.spec.ts [hunk 4]
```diff
@@ -0,0 +1,467 @@
+import * as prompts from '@clack/prompts';
+import fs, { PathLike } from 'fs';
+import path from 'path';
+import { expect, it, vi } from 'vitest';
```
Comment (supproduction) [inline libs/template/cli/src/commands/create.spec.ts]: We use global in tscofig. So exports are not necessary. I think same for other files

File: libs/template/cli/src/commands/create.spec.ts [hunk 5]
```diff
@@ -0,0 +1,467 @@
...
+    get = vi.fn();
+  }
+
+  class MockNodeUtilService {
+    downloadFile = vi.fn();
+    extractZip = vi.fn();
+    copyFolder = vi.fn();
+  }
+
+  function setup() {
```
Comment (supproduction) [inline libs/template/cli/src/commands/create.spec.ts:21]: So we agreed using setup everywhere?

**Review by supproduction — CHANGES_REQUESTED** [review]

File: libs/platform/core/src/services/feature-options/default-feature-options.service.ts (hunk 1 above)
Comment (gund) [reply libs/platform/core/src/services/feature-options/default-feature-options.service.ts:17]: We can't remove that overload version as it's not going to be compatible with the generic overload and implementation.
image

File: libs/template/cli/bin/cli.spec.ts (hunk 2 above)
Comment (gund) [reply libs/template/cli/bin/cli.spec.ts:1]: `vi` is not global, if I remove it I get `Cannot find name 'vi'.` so it should be imported.
Also global typings are from jest which are not quite correct so I also needed to import `expect` to make sure it's properly typed.

File: libs/template/cli/src/commands/create.spec.ts (hunk 4 above)
Comment (gund) [reply libs/template/cli/src/commands/create.spec.ts]: I need at least `vi` and `expect` imports for the typings to be correct, `it` can be removed.

File: libs/template/cli/src/commands/create.spec.ts (hunk 5 above)
Comment (gund) [reply libs/template/cli/src/commands/create.spec.ts:21]: It's been using in the i18n and auth so I think we are ok with using this style and beforeEach?

**Review by supproduction — DISMISSED** [review]

**Review by supproduction — APPROVED** [review]

**Pull request merged**